share/completions/mkdir.fish

```diff
@@ -1,9 +1,20 @@
-complete -c mkdir -s m -l mode --description 'Set file mode (as in chmod)' -x
-complete -c mkdir -s p -l parents --description 'Make parent directories as needed'
-complete -c mkdir -s v -l verbose --description 'Print a message for each created directory'
-complete -c mkdir -s Z --description 'Set SELinux security context of each created directory to the default type'
-complete -c mkdir -l context --description 'Like -Z' -f
-complete -c mkdir -l help --description 'Display help'
-complete -c mkdir -l version --description 'Output version'
 
+# Checks if we are using GNU tools
+if mkdir --version > /dev/null ^ /dev/null
+    complete -c mkdir -l version --description 'Output version'
+    complete -c mkdir -s m -l mode --description 'Set file mode (as in chmod)' -x	
+    complete -c mkdir -s p -l parents --description 'Make parent directories as needed'
+    complete -c mkdir -s v -l verbose --description 'Print a message for each created directory'
+    complete -c mkdir -l help --description 'Display help'
 
+else
+    complete -c mkdir -s m --description 'Set file mode (as in chmod)' -x
+    complete -c mkdir -s p --description 'Make parent directories as needed'
+    complete -c mkdir -s v --description 'Print a message for each created directory'
+end
+
+# Checks if SELinux is installed
+if command -s sestatus > /dev/null ^ /dev/null
+    complete -c mkdir -s Z --description 'Set SELinux security context of each created directory to the default type'
+    complete -c mkdir -l context --description 'Like -Z' -f
```

Comment:
I see that  -Z is the short option for --context -- so instead of giving it a description of "Like -Z", just let it have the same description and have fish show the options together .